src/features/unified-share-modal/__tests__/UnifiedShareForm.test.js

```diff
@@ -243,6 +243,7 @@ describe('features/unified-share-modal/UnifiedShareForm', () => {
             expect(wrapper.exists('.upsell-inline-notice')).toBe(false);
             expect(wrapper.exists('.upsell-title')).toBe(false);
             expect(wrapper.exists('.upsell-body')).toBe(false);
+            expect(wrapper.exists('UpgradeBadge')).toBe(false);
```

Comment:
I think we need to cover all use cases for the possible options of `showUpgradeOptions`, `showUpgradeInlineNotice` and `upsellInlineNotice` which would decide to render the upgradeLinkDescription. Probably easier to use a `test.each` to loop through the possible options.